diff --git a/src/Plugin/EntityReferenceSelection/ParagraphSelection.php b/src/Plugin/EntityReferenceSelection/ParagraphSelection.php
index fb4a04d..d234a6b 100644
--- a/src/Plugin/EntityReferenceSelection/ParagraphSelection.php
+++ b/src/Plugin/EntityReferenceSelection/ParagraphSelection.php
@@ -277,6 +277,33 @@ class ParagraphSelection extends DefaultSelection {
     if (isset($match) && $label_key = $entity_type->getKey('label')) {
       $query->condition($label_key, $match, $match_operator);
     }
+    elseif (!empty($match)) {
+      // On permet d'effectuer la recherche dans un format particulier
+      /**
+       * La recherche peut avoir deux approche :
+       * 1- un texte simple => La recherche se fait sur la colonne type.
+       * 2- un texte commençant par '%%' => dans ce cas c'est constitué du champs, de l'opérateur et de la valeur.
+       * Exemple 1 => %%type=dd:id=12.
+       * Exemple 2 => %%id=125.
+       */
+      if (str_contains($match, "%%")) {
+        $array = explode("%%", $match);
+        $stringQuery = $array[1];
+        $fields = explode(":", $stringQuery);
+        foreach ($fields as $field) {
+          if (str_contains($field, "=")) {
+            [
+              $field_name,
+              $value
+            ] = explode("=", $field);
+            $query->condition($field_name, $value);
+          }
+        }
+      }
+      else {
+        $query->condition("type", $match, $match_operator);
+      }
+    }
 
     // Add entity-access tag.
     $query->addTag($target_type . '_access');
